src/hotspot/share/oops/constantPool.cpp

```diff
@@ -1331,6 +1358,7 @@ bool ConstantPool::compare_entry_to(int index1, const constantPoolHandle& cp2,
     int recur1 = uncached_klass_ref_index_at(index1);
     int recur2 = cp2->uncached_klass_ref_index_at(index2);
     bool match = compare_entry_to(recur1, cp2, recur2);
+    match |= is_unresolved_class_mismatch(recur1, cp2, recur2);
```

Comment:
This is changing the definition of a "match" such that in all cases an unresolved class entry is considered equal to a resolved class entry - but only for these Field and MethodRefs. I don't see how this connects back to the original problem. Nor do I see why this is alway a correct thing to do.

Reply:
I agree that it may be not the right place to call is_unresolved_class_mismatch method because we capture all cases calling compare_entry_to. For a JVM_CONSTANT_Class, compare_entry_to is recovered by the call to is_unresolved_class_mismatch in VM_RedefineClasses::merge_constant_pools. The idea is to do the same for other reference to the class in FieldRefs and (Interface)MethodRefs because otherwise the entry does not match with pre-resolved Throwable class.